staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_discovery.go

```diff
@@ -47,10 +48,16 @@ func (r *versionDiscoveryHandler) ServeHTTP(w http.ResponseWriter, req *http.Req
 		return
 	}
 
+	clusterName, err := genericapirequest.ClusterNameFrom(req.Context())
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
 	requestedGroup := pathParts[1]
 	requestedVersion := pathParts[2]
 
-	crds, err := r.crdLister.List(req.Context(), labels.Everything())
+	crds, err := r.crdLister.Cluster(clusterName).List(req.Context(), labels.Everything())
```

Comment:
Doing this means we can eventually remove ctx from List() if we feel like it. That's the only reason it's there for now.